ape_foundry/provider.py

```diff
@@ -489,6 +490,15 @@ def relock_account(self, address: AddressType):
         if address in self.account_manager.test_accounts._impersonated_accounts:
             del self.account_manager.test_accounts._impersonated_accounts[address]
 
+    def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
+        if txn.sender and txn.sender in self.unlocked_accounts:
+            balance = self.get_balance(txn.sender)
+            if balance == 0:
+                # Hack to allow impersonating contracts without funds easier.
+                self.set_balance(txn.sender, self.conversion_manager.convert("10000 ether", int))
```

Comment:
maybe this is not the right fix, but basically i have a script that works on brownie with anvil but fails with ape because the account does not have enough funds. on brownie, the account also doesnt have enough funds though... so im not really sure what is going on or what to do...

Reply:
Tried to do a little legwork for you but didn't find out anything.  Brownie doesn't appear do anything explicit when unlocking other than calling the RPC method.  And hardhat doesn't document said behavior.  :shrug:

Reply:
apparently it is because brownie uses type 0 by default. when using type 2, it is the same
thus, am deleting this part and will instruct users to either use type=0 for set a balance on their unlocked accounts.
either is super easy to do!